### ISSUE TITLE
[MLv2] [BE] [Bug] fieldValuesSearchInfo issues with native queries

### ISSUE THREAD

**Issue by ranquild** [metabase/metabase#37100]
2 E2E tests were skipped in https://github.com/metabase/metabase/pull/37072 because of issues in `fieldValuesSearchInfo`:

- joins.cy.spec.js - `should join a native question`
- native.cy.spec.js - `shouldn't remove rows containing NULL when using "Is not" or "Does not contain" filter (metabase#13332)`

`fieldValuesSearchInfo` in these tests return `fieldId` and `searchFieldId` as strings, but it should return only real field ids (numbers). If there is no field id it should return `null` `fieldId` and `searchFieldId`. 

In other words, for native queries `fieldValuesSearchInfo` should return `null` for `fieldId` and `searchFieldId` unless it's a native model with column metadata overrides where a DB field is set; in that case the same logic as for MBQL queries should work.

Please unskip the tests when the issue is fixed.

**Comment by darksciencebase** [comment]
> blocks merging the last FE PR for the filter modal.
